### PR TITLE
Pod outage Chaos scenario

### PR DESCRIPTION
**Pull request by yogananth-subramanian**
Pod outage chaos scenario blocks traffic at pod level irrespective of the network policy used. With the current network policies, it is not possible to explicitly block ports which are enabled by allowed network policy rule. This chaos scenario addresses this issue by using OVS flow rules to block ports related to the pod. It supports OpenShiftSDN and OVNKubernetes based networks.

Below example config blocks access to openshift console.
````
- id: pod_outage
  config: 
    namespace: openshift-console
    direction:
        - ingress
    ingress_ports:
        - 8443
    label_selector: 'component=ui'
````

### CODE REVIEW

**Comment by chaitanyaenr** [comment]
test

File: config/config.yaml [hunk 1]
```diff
@@ -18,6 +18,7 @@ kraken:
             - scenarios/openshift/regex_openshift_pod_kill.yml
             - scenarios/openshift/vmware_node_scenarios.yml
             - scenarios/openshift/network_chaos_ingress.yml
+            - scenarios/openshift/pod_outage.yml
```
Comment (chaitanyaenr) [inline config/config.yaml:21]: s/pod_outage/pod_network_outage to make sure we don't confuse it with pod scenarios. Thoughts?

File: scenarios/openshift/pod_outage.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    direction:                          # Optioinal - List of directions to apply filters 
+        - <egress/ingress>              # Default both egress and ingress
+    ingress_ports:                      # Optional - List of ports to block traffic on
+        - <port number>                 # Default [], i.e. all ports
+    egress_ports:                       # Optional - List of ports to block traffic on
+        - <port number>                 # Default [], i.e. all ports   
+    pod_name: <pod name>                # When label_selector is not specified, pod matching the name will be selected for the chaos scenario
+    label_selector: <label_selector>    # When pod_name is not specified, pod with matching label_selector is selected for chaos scenario
+    instance_count: <number>            # Number of nodes to perform action/select that match the label selector 
+    kubeconfig: <path>                  # Path to kubernetes config file. If not specified, it defaults to ~/.kube/config
```
Comment (chaitanyaenr) [inline scenarios/openshift/pod_outage.yml:14]: If not specified should pick up the config from Kraken config like we discussed

File: config/config.yaml (hunk 1 above)
Comment (yogananth-subramanian) [reply config/config.yaml]: Agreed, changed to pod_network_outage in the latest commit.

File: scenarios/openshift/pod_outage.yml (hunk 2 above)
Comment (yogananth-subramanian) [reply scenarios/openshift/pod_outage.yml]: Agreed, changed the behavior in the latest commit to use the same config used by kraken.

File: docs/pod_network_outages.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+### Pod outage
```
Comment (chaitanyaenr) [inline docs/pod_network_outages.md:1]: s/pod outage/pod network outage

Comment (chaitanyaenr) [reply docs/pod_network_outages.md:1]: Also let's add this scenario to the list -https://github.com/redhat-chaos/krkn#kubernetesopenshift-chaos-scenarios-supported

**Review by chaitanyaenr — APPROVED** [review]
LGTM

File: scenarios/openshift/pod_network_outage.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    ingress_ports:                      # Optional - List of ports to block traffic on
+        - <port number>                 # Default [], i.e. all ports
+    egress_ports:                       # Optional - List of ports to block traffic on
+        - <port number>                 # Default [], i.e. all ports
+    pod_name: <pod name>                # When label_selector is not specified, pod matching the name will be selected for the chaos scenario
+    label_selector: <label_selector>    # When pod_name is not specified, pod with matching label_selector is selected for chaos scenario
+    instance_count: <number>            # Number of nodes to perform action/select that match the label selector
+    wait_duration: <time_duration>      # Default is 300. Ensure that it is at least about twice of test_duration
+    test_duration: <time_duration>      # Default is 120
+    kraken_config: <path>               # Specify this if you want to use Cerberus config
```
Comment (paigerube14) [inline scenarios/openshift/pod_network_outage.yml:16]: What is this variable doing? I'm not sure what to set here. The variable says kraken but comment says cerberus?

**Comment by paigerube14** [comment]
Do we need to also edit the plugin.schema.json file here? I  am getting this error when trying to edit the scenario file
Screenshot 2023-05-02 at 4 13 20 PM (2)

File: scenarios/openshift/pod_network_outage.yml (hunk 4 above)
Comment (yogananth-subramanian) [reply scenarios/openshift/pod_network_outage.yml:16]: kraken_config is the path to kraken config file, to publish kraken status to cerberus and also to get go no-go signal from cerberus

Comment (chaitanyaenr) [reply scenarios/openshift/pod_network_outage.yml:16]: @yogananth-subramanian and I discussed leveraging the required params from kraken config and try removing this parameter.

Comment (yogananth-subramanian) [reply scenarios/openshift/pod_network_outage.yml]: @paigerube14 @chaitanyaenr removed the requirement to explicitly pass kraken config file in commit https://github.com/redhat-chaos/krkn/commit/ff1a489cd6cca3e47c5967b849d1189de0e591aa. The plugin automatically passes the kraken config file , just like kubeconfig file.

**Comment by yogananth-subramanian** [comment]
> Do we need to also edit the plugin.schema.json file here? I am getting this error when trying to edit the scenario file Screenshot 2023-05-02 at 4 13 20 PM (2)
Regenerated the plugin.schema.json file, it should be resolved now.

**Review by chaitanyaenr — APPROVED** [review]
LGTM!

**Review by paigerube14 — APPROVED** [review]
/lgtm

**Comment by chaitanyaenr** [comment]
@yogananth-subramanian unit tests are failing which is blocking the PR merge: https://github.com/redhat-chaos/krkn/actions/runs/4946862575/jobs/8845486738?pr=398.

**Pull request merged**